packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -356,6 +356,13 @@ export class FluidDataStoreRuntime
 	public async request(request: IRequest): Promise<IResponse> {
 		try {
 			const parser = RequestParser.create(request);
+			// if there are not path parts, and the request is via a handle
+			// then we should return the entrypoint object for this runtime.
+			// this allows the entrypoint handle to be resolved without the need
+			// for the entrypoint object to know anything about requests or handles.
+			if (parser.pathParts.length === 0 && request.headers?.viaHandle === true) {
+				return { mimeType: "fluid/object", status: 200, value: await this.entryPoint.get() };
```

Comment:
I'm surprised there's no circular get structure call here.

Reply:
its because the entrypoint handle is a fluid object handle, which always has a real reference to the object itself. those get serialized and then deserialized into a plain handle, which really just has a path, the path walk to the runtime, which calls this, and get the true object off the internal fluid object handle.

i'll add this to the comment